src/mscorlib/src/System/Buffer.cs

```diff
@@ -545,47 +538,62 @@ internal unsafe static void Memmove(byte* dest, byte* src, nuint len)
 #if BIT64
                 *(long*)(dest + i) = *(long*)(src + i);
                 *(long*)(dest + i + 8) = *(long*)(src + i + 8);
-#else
+#else // BIT64
                 *(int*)(dest + i) = *(int*)(src + i);
                 *(int*)(dest + i + 4) = *(int*)(src + i + 4);
                 *(int*)(dest + i + 8) = *(int*)(src + i + 8);
                 *(int*)(dest + i + 12) = *(int*)(src + i + 12);
-#endif
+#endif // BIT64
 
                 i = counter;
-                
+
                 // See notes above for why this wasn't used instead
                 // i += 16;
             }
             while (counter <= end);
 
-            if ((len & 8) != 0)
+            // We have <= 15 bytes to copy at this point.
+
+            if ((mask & 8) != 0)
```

Comment:
Also consider, something like (I think the math is write, but I haven't double checked it)
First Write,
`nuint i = 8u - (dest % sizeof(nuint))`
`len -= i;
`nuint r = len % 16`
`nuint l = (sizeof(nuint) - r);
Loop
Final
```
	cases 13-15:
		*(int*)(dest + i) = *(int*)(src + i)
		*(int*)(dest + i + 4) = *(int*)(src + i + 4)
		*(int*)(dest + i + 8 - l) = *(int*)(src + i + 8 - l)
	case 12:
		*(int*)(dest + i) = *(int*)(src + i)
		*(int*)(dest + i + 4) = *(int*)(src + i + 4)
		*(int*)(dest + i + 8) = *(int*)(src + i + 8)
	cases 9-11:
		*(int*)(dest + i) = *(int*)(src + i)
		*(int*)(dest + i + 4 - l) = *(int*)(src + i + 4 - l)
	case 8:
		*(int*)(dest + i) = *(int*)(src + i)
		*(int*)(dest + i + 4) = *(int*)(src + i + 4)
	cases 5-7:
		*(int*)(dest + i) = *(int*)(src + i)
		*(int*)(dest + i + 4 - l) = *(int*)(src + i + 4 - l)
	case 4:
		*(int*)(dest + i) = *(int*)(src + i)
	cases 1-3
		*(int*)(dest + i -l) = *(int*)(src + i - l)
	case 0:
		return
```